src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ISymbolExtensions.cs

```diff
@@ -749,5 +749,15 @@ private static bool VerifyGetAsyncEnumerator(IMethodSymbol getAsyncEnumerator)
         public static bool IsSymbolWithSpecialDiscardName(this ISymbol symbol)
             => symbol.Name.StartsWith("_") &&
                (symbol.Name.Length == 1 || uint.TryParse(symbol.Name.Substring(1), out _));
+
+        /// <summary>
+        /// Returns <see langword="true"/>, if the symbol is marked with the <see cref="System.ObsoleteAttribute"/>.
+        /// </summary>
+        /// <param name="symbol"></param>
+        /// <returns><see langword="true"/> if the symbol is marked with the <see cref="System.ObsoleteAttribute"/>.</returns>
+        public static bool IsObsolete(this ISymbol symbol)
+            => symbol.GetAttributes().Any(x =>
+                x.AttributeClass?.MetadataName == nameof(ObsoleteAttribute) &&
+                x.AttributeClass?.ContainingNamespace.Name == nameof(System));
```

Comment:
`x.AttributeClass is { MetadataName: nameof(ObsoleteAttribute), ContainingNamespace: nameof(System) };`